plugins/scanners/scanoss/src/main/kotlin/ScanOss.kt

```diff
@@ -101,4 +110,86 @@ class ScanOss(
+        val removeRules: List<RemoveRule>
+    )
+
+    private fun buildSettingsFromORTContext(context: ScanContext): ScanossSettings {
+        val rules = processSnippetChoices(context.snippetChoices)
+        val bom = Bom.builder()
+            .ignore(rules.ignoreRules)
+            .include(rules.includeRules)
+            .replace(rules.replaceRules)
+            .remove(rules.removeRules)
+            .build()
+        return ScanossSettings.builder().bom(bom).build()
+    }
+
+    fun processSnippetChoices(snippetChoices: List<SnippetChoices>): ProcessedRules {
+        val includeRules = mutableListOf<Rule>()
+        val ignoreRules = mutableListOf<Rule>()
+        val replaceRules = mutableListOf<ReplaceRule>()
+        val removeRules = mutableListOf<RemoveRule>()
+
+        snippetChoices.forEach { snippetChoice ->
+            snippetChoice.choices.forEach { choice ->
+                when (choice.choice.reason) {
+                    SnippetChoiceReason.ORIGINAL_FINDING -> {
+                        processOriginalFinding(
+                            choice = choice,
+                            includeRules = includeRules
+                        )
+                    }
+
+                    SnippetChoiceReason.NO_RELEVANT_FINDING -> {
+                        processNoRelevantFinding(
+                            choice = choice,
+                            removeRules = removeRules
+                        )
+                    }
+
+                    SnippetChoiceReason.OTHER -> {
+                        processOtherReason(choice)
+                    }
+                }
+            }
+        }
+
+        return ProcessedRules(includeRules, ignoreRules, replaceRules, removeRules)
+    }
+
+    private fun processOriginalFinding(choice: SnippetChoice, includeRules: MutableList<Rule>) {
+        includeRules.add(
+            Rule.builder()
+                .purl(choice.choice.purl)
+                .path(choice.given.sourceLocation.path)
+                .build()
+        )
+    }
+
+    private fun processNoRelevantFinding(choice: SnippetChoice, removeRules: MutableList<RemoveRule>) {
+        val builder = RemoveRule.builder()
+
+        builder.path(choice.given.sourceLocation.path)
+
+        // Set line range only if both start and end lines are positive numbers
+        // If either line is < 0, no line range is set and the rule will apply to the entire file
+        if (choice.given.sourceLocation.startLine > 0 && choice.given.sourceLocation.endLine > 0) {
+            builder.startLine(choice.given.sourceLocation.startLine)
+            builder.endLine(choice.given.sourceLocation.endLine)
+        }
+
+        val rule = builder.build()
+        removeRules.add(rule)
+    }
+
+    private fun processOtherReason(snippetChoice: SnippetChoice) {
+        logger.info {
+            "Encountered OTHER reason for snippet choice in file ${snippetChoice.given.sourceLocation.path}"
+        }
+    }
```

Comment:
Should this be inlined at the only call site to not create the impression that it does not than logging?